state/state.go

```diff
@@ -202,6 +220,7 @@ func (i *Indexer) getVerifiedParent(ctx context.Context, hash []byte, height uin
 				"Got unexpected header value for block at height %d: expected %x, got %x",
 				height, hash, hdr.Id,
 			)
+			skipCache = true
```

Comment:
We `skipCache` as we got invalid data from an Access API server, i.e. the `GetBlockHeaderByHeight` call for a particular block ID did not have a matching block ID within the response.

Since this response would have been cached the first time round, we want to bypass the cache when we call it again, otherwise it would just keep returning the same data.

By setting `skipCache = true`, this will then leverage `cache.Skip` when it re-enters the loop in the future, and thus bypass the cache for the Access API calls.

We only care about `skipCacheWarned` in the block of code that sets up `cache.Skip`, i.e. to warn about the first time that we're bypassing the cache.